### PR TITLE
Add jwt.Credentials.from_signing_credentials, remove serivce_account.Credentials.to_jwt_credentials

### PR DESCRIPTION
**Pull request by jonparrott**
This change made possible by making `Signing.signer` public.

### CODE REVIEW

File: google/auth/jwt.py [hunk 1]
```diff
@@ -371,6 +371,40 @@ def from_service_account_file(cls, filename, **kwargs):
...
+
+        The new instance will use the same signer as the existing instance and
+        will use the existing instance's signer email as the issuer and
+        subject by default.
+
+        Example::
+
+            svc_creds = service_account.Credentials.from_service_account_file(
+                'service_account.json')
+            jwt_ceds = jwt.Credentials.from_signing_credentials(
```
Comment (dhermes) [inline google/auth/jwt.py]: s/`ceds`/`creds`

File: google/auth/jwt.py [hunk 2]
```diff
@@ -371,6 +371,40 @@ def from_service_account_file(cls, filename, **kwargs):
...
+                svc_creds,
+                audience='https://speech.googleapis.com')
```
Comment (dhermes) [inline google/auth/jwt.py]: Didn't you discover in your system tests that this is a bad audience?

Comment (jonparrott) [inline google/auth/jwt.py]: Yeah, fixed throughout.

File: google/auth/jwt.py (hunk 1 above)
Comment (jonparrott) [inline google/auth/jwt.py]: Fixed.

**Comment by dhermes** [comment]
LGTM

**Pull request merged**